language/libra-vm/src/libra_vm.rs

```diff
@@ -287,14 +287,20 @@ impl LibraVM {
             warn!("[VM] Custom scripts not allowed: {:?}", &script.code());
             return Err(VMStatus::new(StatusCode::UNKNOWN_SCRIPT));
         };
-        self.run_prologue(&mut ctx, &txn_data)?;
+        let meta_ty_args = script
+            .meta_ty_args()
+            .iter()
+            .map(|tag| self.resolve_type_argument(&mut ctx, tag))
+            .collect::<VMResult<Vec<_>>>()?;
+        self.run_prologue(&mut ctx, meta_ty_args.clone(), &txn_data)?;
```

Comment:
I don't think there is a use-case for passing arguments to the prologue/epilogue other than `gas_specifier`?
Thus, I think what you actually want to do is something like:
(1) Add a `gas_specifier` field to `TransactionMetadata`
(2) Here, 
```
let prologue_type_args = vec![txn_data.gas_specifier];
self.run_prologue(&mut ctx, prologue_type_args, &txn_data)?;
```

and similar for  the epilogue.

Reply:
I agree. My only question is whether or not we should add the `gas_specifier` to the `TransactionMetadata` -- it can't be accessed in Move (since it's a Move type).

Reply:
I think the `gas_specifier` can be part of the `TransactionMetadata`. The `TransactionMetadata` is essentially a bunch of arguments to the epilogue and prologue, and it seems fine to me to include type arguments here. @tzakian I'd say it's still accessible in Move, in the form of an instantiated type parameter.